wallet/src/de/schildbach/wallet/ui/dashpay/DashPayViewModel.kt

```diff
@@ -56,4 +72,60 @@ class DashPayViewModel(application: Application) : AndroidViewModel(application)
         emit(Resource.loading(null))
         emit(platformRepo.isPlatformAvailable())
     }
+    // username registration functions
+
+    val createUsernameLiveData = Transformations.switchMap(registerUsernameLiveData) { usernameInfo ->
+        val wallet = walletApplication.wallet
+        liveData(Dispatchers.IO) {
```

Comment:
We should probably use a wider scope for this operation, I need to think of it.

Reply:
if we update the database with the identity creation status, then do we need to have the createUsernameLiveData in this view model?

What if we created a thread or used a coroutine to run the createUserName process.  As the process proceeds through each step, we update the database with the current status.  Then views (such as the home screen or others) subscribe to that database.  As the database changes, the views are updated?

Is that what you mean by scope?

Reply:
Not exactly, I was rather referring to the coroutines scope (https://developer.android.com/topic/libraries/architecture/coroutines). In the current implementation we use `livedata` scope so the coroutine is only active as long as the createUsernameLiveData is observed by at least one observer (ie CreateUsernameActivity) if you for instance move to the to background before finishing the job it can be suspended or even canceled by the system.
Initially I was thinking about using an `GlobalScope.launch` or something similar but finally I came to the conclusion that the identity creating should be treated as long-running operation and managed by WorkManager (https://developer.android.com/topic/libraries/architecture/workmanager/basics).
I'm working on transferring your code to the Worker.

The rest remains as you described: the process (managed by WorkManager) proceeds through each step, we update the database with the current status. Then views subscribe to that database. As the database changes, the views are updated.

Reply:
A perfect thing for our use case provided by WorkManager is chained tasks feature. We can define a chain of tasks that has to be completed in order to create identity and the WorkManager will take care of correct execution of each task handling loses oh connectivity etc https://developer.android.com/topic/libraries/architecture/workmanager/advanced

Reply:
That sounds great.